ligninkmc/kmc_common.py

```diff
@@ -320,8 +418,25 @@ def __hash__(self):
         #     "By default, the __hash__() values of str and bytes objects are “salted” with an unpredictable random
         #     value. Although they remain constant within an individual Python process, they are not predictable
         #     between repeated invocations of Python."
-        key_as_num = sum([ord(x) % 32 for x in self.key])
-        return key_as_num + sum(self.index) * 1000 + int(self.rate * 10000)
+        # Original hash method, which did not allow testing because of random salting:
+        #    return hash((tuple(self.index), self.key, self.bond))
+        # Replacement hash method which gave consistent results, but unexpected changed outcomes
+        #    key_as_num = sum([ord(x) % 32 for x in self.key])
+        #    return key_as_num + sum(self.index) * 1000 + int(self.rate * 10000)
+        # The bash below is repeatable and gives similar results to the original hash. It is not directly invoked
+        #     in the package (unsettling that the hash method used for dictionary keys would change results), but it
+        #     is left in case there is any behind the scenes hashing
+        if not self.index:
+            index_join = 0
+        else:
+            index_join = int("".join([str(x) for x in self.index]))
+        index_bytes = index_join.to_bytes((index_join.bit_length() + 7) // 8, 'big')
+        key_bytes = self.key.encode()
+        bond_list_str = "".join([str(x) for x in self.bond])
+        bond_list_bytes = bond_list_str.encode()
+        event_bytes = b''.join([index_bytes, key_bytes, bond_list_bytes])
+        # the hash call below "right-sizes" the value
+        return hash(int.from_bytes(event_bytes, 'big'))
```

Comment:
I have already set up such tests and shown that the results are sensitive to hashing. As noted, this hashing method gives similar results to your original hash, but allows for the repeatability required for testing.

As noted, I left this in place in case python does any behind-the-scenes hashing, but removed the explicit hash calls which are not required and could lead to different results with different hashing methods.